### ISSUE TITLE
NPE in com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getDocumentationFor

### ISSUE THREAD

**Issue by lukehutch** [ceylon/ceylon-ide-eclipse#1659]
I get numerous NullPointerExceptions in the log for the Eclipse Ceylon IDE. (One of them popped up with an Eclipse warning dialog, although most of the time no warning is given.)

```
!ENTRY org.eclipse.ui.workbench 4 2 2015-12-20 00:44:16.822
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.ui.workbench".
!STACK 0
java.lang.NullPointerException
        at com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getDocumentationFor(DocumentationHover.java:1366)
        at com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getDocumentationHoverText(DocumentationHover.java:881)
        at com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getHoverText(DocumentationHover.java:429)
        at com.redhat.ceylon.eclipse.code.hover.DocumentationView.update(DocumentationView.java:237)
        at com.redhat.ceylon.eclipse.code.editor.MarkOccurrencesAction.recomputeAnnotationsForSelection(MarkOccurrencesAction.java:297)
        at com.redhat.ceylon.eclipse.code.editor.MarkOccurrencesAction.setUpActiveEditor(MarkOccurrencesAction.java:521)
        at com.redhat.ceylon.eclipse.code.editor.MarkOccurrencesAction.access$000(MarkOccurrencesAction.java:51)
        at com.redhat.ceylon.eclipse.code.editor.MarkOccurrencesAction$EditorPartListener.partActivated(MarkOccurrencesAction.java:105)
        at org.eclipse.ui.internal.WorkbenchPage$16.run(WorkbenchPage.java:4861)
        at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
        at org.eclipse.ui.internal.WorkbenchPage.firePartActivated(WorkbenchPage.java:4858)
        at org.eclipse.ui.internal.WorkbenchPage.access$18(WorkbenchPage.java:4846)
        at org.eclipse.ui.internal.WorkbenchPage$E4PartListener.partActivated(WorkbenchPage.java:210)
        at org.eclipse.e4.ui.internal.workbench.PartServiceImpl$3.run(PartServiceImpl.java:241)
        at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
        at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.firePartActivated(PartServiceImpl.java:238)
        at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:692)
        at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:620)
        at org.eclipse.e4.ui.internal.workbench.swt.AbstractPartRenderer.activate(AbstractPartRenderer.java:106)
        at org.eclipse.e4.ui.workbench.renderers.swt.ContributedPartRenderer$1.handleEvent(ContributedPartRenderer.java:61)
        at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
        at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4454)
        at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1388)
        at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1412)
        at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1397)
        at org.eclipse.swt.widgets.Shell.setActiveControl(Shell.java:1718)
        at org.eclipse.swt.widgets.Control.gtk_button_press_event(Control.java:3048)
        at org.eclipse.swt.widgets.Control.gtk_button_press_event(Control.java:2984)
        at org.eclipse.swt.widgets.Composite.gtk_button_press_event(Composite.java:715)
        at org.eclipse.swt.widgets.Canvas.gtk_button_press_event(Canvas.java:154)
        at org.eclipse.swt.widgets.Widget.windowProc(Widget.java:2077)
        at org.eclipse.swt.widgets.Control.windowProc(Control.java:5534)
        at org.eclipse.swt.widgets.Display.windowProc(Display.java:4668)
        at org.eclipse.swt.internal.gtk.OS._gtk_main_do_event(Native Method)
        at org.eclipse.swt.internal.gtk.OS.gtk_main_do_event(OS.java:9106)
        at org.eclipse.swt.widgets.Display.eventProc(Display.java:1253)
        at org.eclipse.swt.internal.gtk.OS._g_main_context_iteration(Native Method)
        at org.eclipse.swt.internal.gtk.OS.g_main_context_iteration(OS.java:2477)
        at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3407)
        at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$9.run(PartRenderingEngine.java:1151)
        at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
        at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1032)
        at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:148)
        at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:636)
        at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:332)
        at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:579)
        at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:150)
        at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:135)
        at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:648)
        at org.eclipse.equinox.launcher.Main.basicRun(Main.java:603)
        at org.eclipse.equinox.launcher.Main.run(Main.java:1465)
        at org.eclipse.equinox.launcher.Main.main(Main.java:1438)

!ENTRY org.eclipse.jface.text 4 0 2015-12-20 00:44:17.212
!MESSAGE Unexpected runtime error while computing a text hover
!STACK 0
java.lang.NullPointerException
        at com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getDocumentationFor(DocumentationHover.java:1366)
        at com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getDocumentationHoverText(DocumentationHover.java:881)
        at com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getHoverText(DocumentationHover.java:429)
        at com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getHoverInfo2(DocumentationHover.java:334)
        at com.redhat.ceylon.eclipse.code.hover.DocumentationHover.getHoverInfo2(DocumentationHover.java:126)
        at com.redhat.ceylon.eclipse.code.hover.BestMatchHover.getHoverInfo2(BestMatchHover.java:109)
        at com.redhat.ceylon.eclipse.code.hover.BestMatchHover.getHoverInfo2(BestMatchHover.java:85)
        at org.eclipse.jface.text.TextViewerHoverManager$4.run(TextViewerHoverManager.java:166)
```

**Comment by gavinking** [comment]
@davidfestal is it normal that `CeylonUnit.getPhasedUnit()` can return null?

If so, it looks like this is the result of your fix for #1633.

**Comment by gavinking** [comment]
This is fixed, probably.
